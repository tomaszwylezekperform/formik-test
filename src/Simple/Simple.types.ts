export interface ISimpleFormValues {
  email: string;
}

export interface ISimpleProps {}
