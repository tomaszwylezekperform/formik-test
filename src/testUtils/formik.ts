import {FormikActions, FormikHandlers, FormikProps} from "formik";

export function getMockedFormikConfig<T>(options: Partial<FormikProps<T>> = {}): FormikProps<T> {
  return {
    ...getMockedFormikActions(),
    ...getMockedFormikHandles(),
    enableReinitialize: false,
    isValidating: false,
    isSubmitting: false,
    dirty: false,
    isValid: false,
    submitCount: 0,
    registerField: jest.fn(),
    unregisterField: jest.fn(),
    values: {} as T,
    errors: {} as T,
    initialValues: {} as T,
    touched: {} as T,
    ...options
  }
}

export function getMockedFormikActions<V>(): FormikActions<V> {
  return {
    setStatus: jest.fn(),
    setError: jest.fn(),
    setErrors: jest.fn(),
    setSubmitting: jest.fn(),
    setTouched: jest.fn(),
    setValues: jest.fn(),
    setFieldValue: jest.fn(),
    setFieldError: jest.fn(),
    setFieldTouched: jest.fn(),
    validateForm: jest.fn(),
    validateField: jest.fn(),
    resetForm: jest.fn(),
    submitForm: jest.fn(),
    setFormikState: jest.fn(),
  }
}

export function getMockedFormikHandles(): FormikHandlers {
  return {
    handleSubmit: jest.fn(),
    handleReset: jest.fn(),
    handleBlur: jest.fn(),
    handleChange: jest.fn(),
  }
}
