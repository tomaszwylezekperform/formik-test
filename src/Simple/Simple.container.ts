import {FormikBag, withFormik} from "formik";
import * as Yup from "yup";
import { Simple} from "./Simple";
import {ISimpleFormValues, ISimpleProps} from "./Simple.types";

export const handleSubmit = (values: ISimpleFormValues, {setSubmitting, }: FormikBag<ISimpleProps, ISimpleFormValues>) => {
  setTimeout(() => {
    setSubmitting(false);

  }, 2000);
};


export const SimpleContainer =  withFormik<ISimpleProps, ISimpleFormValues>({
  displayName: 'SimpleForm',
  handleSubmit,
  mapPropsToValues: (props: ISimpleProps): ISimpleFormValues => ({
    email: ''
  }),
  // validate: (values: IFormHomeValues): FormikErrors<IFormHomeValues> => {
  //   const errors: FormikErrors<IFormHomeValues> = {};
  //   if (!values.email) {
  //     errors.email = 'Required';
  //   } else if (/\S+@\S+\.\S+/.test(values.email)) {
  //     errors.email = 'Invalid email addres';
  //   }
  //   return errors;
  // }, OR
  validationSchema: Yup.object().shape({
    email: Yup.string().email('Invalid email addres').required('Required'),
  })
})(Simple)
