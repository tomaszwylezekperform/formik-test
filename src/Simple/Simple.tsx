import * as React from "react";

import {Field, FormikProps } from 'formik';
import {InputComponent} from "../shared/InputComponent";
import {ISimpleFormValues, ISimpleProps} from "./Simple.types";


export class Simple extends React.Component<ISimpleProps & FormikProps<ISimpleFormValues>, {}> {
  public render() {

    const {
      dirty,
      isSubmitting,
      handleSubmit,
      handleReset,
    } = this.props;

    return (
      <div>
        <h1>Simple form</h1>
        <form onSubmit={handleSubmit}>
          <Field component={InputComponent} name="email" label="email"/>
          <button
            type="button"
            className="outline"
            onClick={handleReset}
            disabled={!dirty || isSubmitting}
          >
            Reset
          </button>
          <button type="submit" disabled={isSubmitting}>
            Submit
          </button>
        </form>
      </div>
    )
  }
}
