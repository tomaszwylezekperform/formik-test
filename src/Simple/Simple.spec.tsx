import {shallow} from "enzyme";
import * as React from 'react';
import {Simple} from './Simple';
import {FormikProps} from "formik";
import {ISimpleFormValues} from "./Simple.types";
import {getMockedFormikConfig} from "../testUtils/formik";

describe('Simple component', () => {
  it('should have basic snapshoot', () => {
    const options: FormikProps<ISimpleFormValues> = getMockedFormikConfig<ISimpleFormValues>({
      errors: {
        email: ''
      },
      initialValues: {
        email: ''
      },
      touched: {
        email: false
      },
      values: {
        email: ''
      },
    });

    const component = shallow(<Simple {...options}/>);

    component.find('form').simulate('submit');
    expect(component).toMatchSnapshot();
  });
});
