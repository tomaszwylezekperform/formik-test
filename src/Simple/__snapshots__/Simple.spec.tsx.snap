// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple component should have basic snapshoot 1`] = `
<div>
  <h1>
    Simple form
  </h1>
  <form
    onSubmit={[Function]}
  >
    <C
      component={[Function]}
      label="email"
      name="email"
    />
    <button
      className="outline"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Reset
    </button>
    <button
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
